schema/integer.go

```diff
@@ -22,11 +22,15 @@ func (v Integer) parse(value interface{}) (interface{}, error) {
 			value = v
 		}
 	}
-	i, ok := value.(int)
-	if !ok {
-		return nil, errors.New("not an integer")
+
+	if i, ok := value.(int); ok {
+		return i, nil
 	}
-	return i, nil
+	if i, ok := value.(int64); ok {
+		return i, nil
+	}
```

Comment:
Like tihs:
```go
switch v := value.(type) {
case float64:
	i, frac := math.Modf(v)
	if frac != 0.0 {
		return nil, errors.New("not an integer")
	}
	return int(v), nil
case int64:
	return int(v), nil
case int:
	return v
}
default:
    return nil, errors.New("not an integer")
```

This always return an `int`. Either we should always return an `int, or we should always return an `int64`. Not sometimes `int` and sometimes `int64`.